gqcp/include/Basis/Transformations/SimpleOrbitalRotationGenerators.hpp

```diff
@@ -0,0 +1,149 @@
+// along with GQCG-GQCP.  If not, see <http://www.gnu.org/licenses/>.
+
+#pragma once
+
+
+#include "Mathematical/Representation/SquareMatrix.hpp"
+
+
+namespace GQCP {
+
+
+/**
+ *  MARK: SimpleOrbitalRotationGenerator implementation
+ */
+
+
+/**
+ *  An orbitalRotationGenerator that can be represented by a single vector.
+ *   
+ *  This class is used as a base class for `ROrbitalRotationGenerator` and `GOrbitalRotationGenerator`, since they are both expressed using a single vector, as opposed to `UOrbitalRotationGenerator`, which uses separate kappa vectors for alpha- and beta- generators. The word 'simple' is used here as an antonym for 'compound'.
+ * 
+ *  @tparam _Scalar                                           The scalar type used for a orbital rotation generator vector: real or complex.
+ *  @tparam _DerivedOrbitalRotationGenerator                  The type of the orbital rotation generator that derives from this class, enabling CRTP and compile-time polymorphism.
+ */
+template <typename _Scalar, typename _DerivedOrbitalRotationGenerators>
+class SimpleOrbitalRotationGenerators {
+
+public:
+    // The scalar type used for a orbital rotation generator vector: real or complex.
+    using Scalar = _Scalar;
+
+    // The type of the orbital rotation generator that derives from this class, enabling CRTP and compile-time polymorphism.
+    using DerivedOrbitalRotationGenerators = _DerivedOrbitalRotationGenerators;
+
+private:
+    // The number of spatial orbitals that can be rotated using these orbital rotation generators.
+    size_t number_of_spatial_orbitals;
+
+    // The strict upper/lower triangle of the kappa matrix.
+    VectorX<Scalar> kappa_vector;
+
+
+public:
+    /**
+     *  MARK: Constructors
+     */
+
+    /**
+     *  Create a `SimpleOrbitalRotationGenerator' from a given kappa vector.
+     * 
+     *  @param  kappa_vector        The orbital rotation generators represented as a vector that corresponds to the strict upper/lower triangle of the kappa matrix.
+     */
+    SimpleOrbitalRotationGenerators(const VectorX<Scalar>& kappa_vector) :
+        kappa_vector {kappa_vector},
+        number_of_spatial_orbitals {strictTriangularRootOf(kappa_vector.size())} {}
+
+
+    /**
+     *  Create a `SimpleOrbitalRotationGenerator' from a given kappa matrix.
+     * 
+     *  @param  kappa_matrix        The orbital rotation generators represented as a vector that corresponds to the full antisymmetric the kappa matrix.
+     */
+    SimpleOrbitalRotationGenerators(const SquareMatrix<Scalar>& kappa_matrix) :
+        SimpleOrbitalRotationGenerators(kappa_matrix.pairWiseStrictReduced()) {}
+
+
+    /**
+     *  MARK: Named Constructors
+     */
+
+    /**
+     *  Construct orbital rotation generators by adding redundant (i.e. 0) occupied-virtual and virtual-virtual generators to the given occupied-occupied generators.
+     * 
+     *  @param o_o_generators       The occupied-occupied orbital rotation generators.
```

Comment:
```suggestion
     *  @param occ_occ_generators       The occupied-occupied orbital rotation generators.
```
Maybe make this consistent with the API name?